diff -u -r -p NVIDIA-Linux-x86_64-384.98.orig/kernel/nvidia-uvm/uvm8_va_block.c NVIDIA-Linux-x86_64-384.98/kernel/nvidia-uvm/uvm8_va_block.c
--- NVIDIA-Linux-x86_64-384.98.orig/kernel/nvidia-uvm/uvm8_va_block.c	2018-01-02 15:44:13.194212337 +0100
+++ NVIDIA-Linux-x86_64-384.98/kernel/nvidia-uvm/uvm8_va_block.c	2018-01-02 15:44:56.495053964 +0100
@@ -36,6 +36,10 @@
 #include "uvm8_perf_prefetch.h"
 #include "uvm8_mem.h"
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,9)
+#include <linux/sched/task_stack.h>
+#endif
+
 typedef enum
 {
     BLOCK_PTE_OP_MAP,
