--- kernel/nvidia/nv-gpu-numa.c.orig	2017-11-22 14:50:37.952637429 +0100
+++ kernel/nvidia/nv-gpu-numa.c	2017-11-22 15:49:21.665327784 +0100
@@ -113,6 +113,8 @@ static NV_STATUS bad_idea_read_string_fr
     struct file *filp;
     int read_count;
 
+    loff_t pos = 0;
+
     filp = filp_open(path_to_file, O_RDONLY, 0);
     if (IS_ERR(filp)) 
     {
@@ -120,7 +122,7 @@ static NV_STATUS bad_idea_read_string_fr
         return NV_ERR_NO_VALID_PATH;
     }
 
-    read_count = kernel_read(filp, 0, read_buffer, read_buffer_size - 1);
+    read_count = kernel_read(filp, (void *) read_buffer, read_buffer_size - 1, &pos);
 
     filp_close(filp, NULL);
 
