--- kernel/nvidia-drm/nvidia-drm-crtc.c.orig	2018-03-02 15:49:41.080997600 +0100
+++ kernel/nvidia-drm/nvidia-drm-crtc.c	2018-03-02 15:49:57.317310484 +0100
@@ -613,7 +613,7 @@ int nv_drm_get_crtc_crc32_ioctl(struct d
         goto done;
     }
 
-    crtc = drm_crtc_find(dev, params->crtc_id);
+    crtc = drm_crtc_find(dev, NULL, params->crtc_id);
     if (!crtc) {
         ret = -ENOENT;
         goto done;
--- kernel/nvidia-drm/nvidia-drm-connector.c.orig	2018-03-02 16:10:57.973611177 +0100
+++ kernel/nvidia-drm/nvidia-drm-connector.c	2018-03-02 16:11:19.966035242 +0100
@@ -87,7 +87,7 @@ static enum drm_connector_status __nv_dr
             break;
         }
 
-        encoder = drm_encoder_find(dev, connector->encoder_ids[i]);
+        encoder = drm_encoder_find(dev, NULL, connector->encoder_ids[i]);
 
         if (encoder == NULL) {
             BUG_ON(encoder != NULL);
