--- kernel/nvidia-uvm/uvm_linux.h	2016/07/22 22:12:51	1.1
+++ kernel/nvidia-uvm/uvm_linux.h	2016/07/22 22:13:04
@@ -563,7 +563,7 @@
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
-static bool radix_tree_empty(struct radix_tree_root *tree)
+static bool radix_tree_is_empty(struct radix_tree_root *tree)
 {
     void *dummy;
     return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
--- kernel/nvidia-uvm/uvm8_gpu.c	2016/07/22 22:13:14	1.1
+++ kernel/nvidia-uvm/uvm8_gpu.c	2016/07/22 22:13:19
@@ -638,7 +638,7 @@
                    gpu->id, uvm_gpu_retained_count(gpu));
 
     // All channels should have been removed before the retained count went to 0
-    UVM_ASSERT(radix_tree_empty(&gpu->instance_ptr_table));
+    UVM_ASSERT(radix_tree_is_empty(&gpu->instance_ptr_table));
 
     // Remove the GPU from the table.
     uvm_spin_lock_irqsave(&g_uvm_global.gpu_table_lock);
