--- nvidia-gfxG04.spec.orig
+++ nvidia-gfxG04.spec
@@ -50,6 +50,7 @@ NoSource:       0
 NoSource:       1
 NoSource:       6
 NoSource:       7
+Patch0:         kernel-4.3-buildfix.patch
 BuildRequires:  kernel-source
 BuildRequires:  kernel-syms
 BuildRequires:  module-init-tools
@@ -120,6 +121,11 @@ echo "kver = %kver"
 %ifarch x86_64
  sh %{SOURCE1} -x
 %endif
+%if %kver >= 40300
+pushd NVIDIA-Linux-x86*-%{version}*/kernel
+%patch0 -p1
+popd
+%endif
 #rm -rf NVIDIA-Linux-x86*-%{version}-*/usr/src/nv/precompiled
 mkdir -p source/%{version}
 rm -rf NVIDIA-Linux-x86*-%{version}*/kernel/uvm
